press/infrastructure/doctype/virtual_disk_resize/virtual_disk_resize.py

```diff
@@ -85,15 +84,23 @@ def after_insert(self):
 			frappe.enqueue_doc(self.doctype, self.name, "run_prerequisites", queue="long", timeout=2400)
 
 	def run_prerequisites(self):
-		self.status = Status.Preparing
-		self.save()
-		self.set_filesystem_attributes()
-		self.set_new_volume_attributes()
-		self.create_new_volume()
-		self.status = Status.Ready
-		self.save()
-		if self.scheduled_time:
-			self.execute()
+		try:
+			self.status = Status.Preparing
+			self.save()
+			self.set_filesystem_attributes()
+			self.set_new_volume_attributes()
+			self.create_new_volume()
+			self.status = Status.Ready
+			self.save()
+			if self.scheduled_time:
+				self.execute()
+		except frappe.QueryTimeoutError:
+			frappe.db.rollback()
```

Comment:
Maybe release the lock as soon as possible via a commit?

Instead of excepting timeout errors?